Address Safer CPP warnings in WebFrame.cpp

kmonsen opened the pull request:
#### 936363b630c3e826040c2ad1b30b7bfd178cfb6e
<pre>
Address Safer CPP warnings in WebFrame.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=290464">https://bugs.webkit.org/show_bug.cgi?id=290464</a>
<a href="https://rdar.apple.com/problem/147948239">rdar://problem/147948239</a>

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::localFrameLoaderClient const):
(WebKit::WebFrame::remoteFrameClient const):
(WebKit::WebFrame::frameLoaderClient const):
(WebKit::WebFrame::page const):
(WebKit::WebFrame::frameTreeData const):
(WebKit::WebFrame::loadDidCommitInAnotherProcess):
(WebKit::WebFrame::commitProvisionalFrame):
(WebKit::WebFrame::removeFromTree):
(WebKit::WebFrame::didReceivePolicyDecision):
(WebKit::WebFrame::startDownload):
(WebKit::WebFrame::convertMainResourceLoadToDownload):
(WebKit::WebFrame::addConsoleMessage):
(WebKit::WebFrame::source const):
(WebKit::WebFrame::selectionAsString const):
(WebKit::WebFrame::size const):
(WebKit::WebFrame::isMainFrame const):
(WebKit::WebFrame::isRootFrame const):
(WebKit::WebFrame::name const):
(WebKit::WebFrame::innerText const):
(WebKit::WebFrame::pendingUnloadCount const):
(WebKit::WebFrame::allowsFollowingLink const):
(WebKit::WebFrame::jsContextForWorld):
(WebKit::WebFrame::jsContextForServiceWorkerWorld):
(WebKit::WebFrame::contentBounds const):
(WebKit::WebFrame::scrollOffset const):
(WebKit::WebFrame::hasVerticalScrollbar const):
(WebKit::WebFrame::containsAnyFormControls const):
(WebKit::WebFrame::jsWrapperForWorld):
(WebKit::WebFrame::setTextDirection):
(WebKit::WebFrame::handleContextMenuEvent):
(WebKit::WebFrame::handleMouseEvent):
(WebKit::WebFrame::frameTextForTesting):

Canonical link: <a href="https://commits.webkit.org/293157@main">https://commits.webkit.org/293157@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dbe910cd92c88df3d5be7d518aed8c61dc555733

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98070 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17701 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7928 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103185 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48599 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17993 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26152 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74663 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31849 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101074 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13629 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88612 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55023 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13412 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48041 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83390 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6627 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105563 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25156 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18319 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83651 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25529 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84793 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83105 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20991 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27744 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5422 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/18790 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25115 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30289 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24935 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28251 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26510 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -162,12 +162,17 @@ Ref<WebFrame> WebFrame::createRemoteSubframe(WebPage& page, WebFrame& parent, We
             opener = openerWebFrame->coreFrame();
     }
 
+    Ref<WebPage> localPage = page;
```
Please omit the `<WebPage>` part here and below.

kmonsen replied:
Done

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -607,14 +612,14 @@ void WebFrame::startDownload(const WebCore::ResourceRequest& request, const Stri
-    auto* localFrame = dynamicDowncast<LocalFrame>(m_coreFrame.get());
-    auto topOrigin = localFrame && localFrame->document() ? std::optional { localFrame->document()->topOrigin().data() } : std::nullopt;
+    RefPtr localFrame = dynamicDowncast<LocalFrame>(m_coreFrame.get());
+    auto topOrigin = localFrame && localFrame->protectedDocument() ? std::optional { localFrame->protectedDocument()->topOrigin().data() } : std::nullopt;
```
The first `protectedDocument()` here is not right. We do not need to protect just to null check.

kmonsen replied:
Sure, to me it doesn't look easy to read when we check one function for null and then use that to call another, you have to know that those two functions represent the same object. Done.

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -623,11 +628,11 @@ void WebFrame::convertMainResourceLoadToDownload(DocumentLoader* documentLoader,
     auto policyDownloadID = *std::exchange(m_policyDownloadID, std::nullopt);
 
-    SubresourceLoader* mainResourceLoader = documentLoader->mainResourceLoader();
+    RefPtr<SubresourceLoader> mainResourceLoader = documentLoader->mainResourceLoader();
```
Please omit the `<X>` part.

kmonsen replied:
Done

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -659,7 +664,7 @@ String WebFrame::source() const
     if (!document)
         return String();
-    TextResourceDecoder* decoder = document->decoder();
+    RefPtr<TextResourceDecoder> decoder = document->decoder();
```
ditto.

kmonsen replied:
Done

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -740,20 +746,23 @@ bool WebFrame::isFrameSet() const
 bool WebFrame::isRootFrame() const
 {
-    return m_coreFrame && m_coreFrame->isRootFrame();
+    RefPtr localCoreFrame = coreFrame();
```
You keep using `local` in the names of local variables but this is confusing, especially for frames where we already have the local vs remote distinction (due to site isolation) and which has a different meaning.

Normally, we'd just use:
```
RefPtr coreFrame = this->coreFrame();
```
or
```
RefPtr frame = coreFrame();
```

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -877,20 +886,27 @@ JSGlobalContextRef WebFrame::jsContextForWorld(DOMWrapperWorld& world)
 JSGlobalContextRef WebFrame::jsContextForWorld(InjectedBundleScriptWorld* world)
 {
-    return jsContextForWorld(world->coreWorld());
+    RefPtr localWorld = world;
```
Even for other things than frame, I don't like the `local` prefix and this is not common practice in our codebase.

cdumez commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -877,20 +886,27 @@ JSGlobalContextRef WebFrame::jsContextForWorld(DOMWrapperWorld& world)
 JSGlobalContextRef WebFrame::jsContextForServiceWorkerWorld(InjectedBundleScriptWorld* world)
 {
-    return jsContextForServiceWorkerWorld(world->coreWorld());
+    RefPtr localWorld = world;
```
We do not need to protect `world` as it is a parameter.

cdumez reviewed (approved):
LGTM

kmonsen committed:
Address Safer CPP warnings in WebFrame.cpp
https://bugs.webkit.org/show_bug.cgi?id=290464
rdar://problem/147948239

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::localFrameLoaderClient const):
(WebKit::WebFrame::remoteFrameClient const):
(WebKit::WebFrame::frameLoaderClient const):
(WebKit::WebFrame::page const):
(WebKit::WebFrame::frameTreeData const):
(WebKit::WebFrame::loadDidCommitInAnotherProcess):
(WebKit::WebFrame::commitProvisionalFrame):
(WebKit::WebFrame::removeFromTree):
(WebKit::WebFrame::didReceivePolicyDecision):
(WebKit::WebFrame::startDownload):
(WebKit::WebFrame::convertMainResourceLoadToDownload):
(WebKit::WebFrame::addConsoleMessage):
(WebKit::WebFrame::source const):
(WebKit::WebFrame::selectionAsString const):
(WebKit::WebFrame::size const):
(WebKit::WebFrame::isMainFrame const):
(WebKit::WebFrame::isRootFrame const):
(WebKit::WebFrame::name const):
(WebKit::WebFrame::innerText const):
(WebKit::WebFrame::pendingUnloadCount const):
(WebKit::WebFrame::allowsFollowingLink const):
(WebKit::WebFrame::jsContextForWorld):
(WebKit::WebFrame::jsContextForServiceWorkerWorld):
(WebKit::WebFrame::contentBounds const):
(WebKit::WebFrame::scrollOffset const):
(WebKit::WebFrame::hasVerticalScrollbar const):
(WebKit::WebFrame::containsAnyFormControls const):
(WebKit::WebFrame::jsWrapperForWorld):
(WebKit::WebFrame::setTextDirection):
(WebKit::WebFrame::handleContextMenuEvent):
(WebKit::WebFrame::handleMouseEvent):
(WebKit::WebFrame::frameTextForTesting):

Canonical link: https://commits.webkit.org/293157@main
